Fix the problem TESTFLAGS is not to be used in Makefile correctly

KentaTada opened the pull request:
Currently, To pass argument "TESTFLAGS" for Makefile to execute "make test" is meaningless because TESTFLAGS is used in both localunittest and localintegration. So, this commit modifies Makefile to handle test targets for "unittest" and "integration" correctly.

Signed-off-by: Kenta Tada <Kenta.Tada@sony.com>

crosbymichael commented:
@KentaTada is the problem that TESTFLAGs is not being passed down or that it's not used in the docker case?

KentaTada commented:
@crosbymichael  I'm sorry I didn't make it clear enough.
The problem is that TESTFLAG is not being passed down to localintegration correctly because TESTFLAG are used in both localunittest and localintegration. For example,  'make test TESTFLAGS="-run=TestValidateNetworkWithoutNETNamespace"' will test 'bats -t tests/integration-run=TestValidateNetworkWithoutNETNamespace' as localintegration test but it will fail. So, I added TESTPATH for localintegration and removed unused TESTFLAG. I think this new flag TESTPATH will be used like 'make test TESTFLAGS="-run=TestValidateNetworkWithoutNETNamespace" TESTPATH="/ps.bats"'.

crosbymichael commented:
LGTM

KentaTada commented:
@crosbymichael  Thank you for your review.

/cc @rjnagal @vmarmol @mrunalp @dqminh @hqhq @cyphar
Please review this commit.

dqminh commented:
LGTM
